src/Reveal.Sdk.Dom.Tests/Data/DataSources/RestDataSourceFixture.cs

```diff
@@ -104,5 +109,156 @@ public void Method_SetsAndGetsValue_WithDifferentInputs(string method)
             Assert.Equal(method, dataSource.Method);
             Assert.Equal(method, dataSource.Properties.GetValue<string>("Method"));
         }
+
+        [Fact]
+        public void RDashDocument_HasCorrectDataSource_WhenLoadFromFile()
+        {
+            // Arrange
+            var filePath = Path.Combine(Environment.CurrentDirectory, "Dashboards", "TestRest.rdash");
+
+            // Act
+            var document = RdashDocument.Load(filePath);
+            var dataSource = document.DataSources.LastOrDefault();
+
+            // Assert
+            Assert.Equal(DataSourceProvider.REST, dataSource.Provider);
+            Assert.NotNull(dataSource.Properties.GetValue<string>("Url"));
+            Assert.NotNull(dataSource.Properties.GetValue<string>("_rpUseAnonymousAuthentication"));
+            Assert.NotNull(dataSource.Properties.GetValue<string>("_rpUsePreemptiveAuthentication"));
+        }
+
+        [Fact]
+        public void ToJsonString_CreatesFormattedJson_ForRestDataSource()
+        {
+            // Arrange
+            var expectedJson =
+                """
+                {
+                  "_type": "DataSourceType",
+                  "Id": "7473e6b8-ba86-4703-8e69-79f83d0742f3",
+                  "Provider": "REST",
+                  "Description": "JSON DS",
+                  "Subtitle": "JSON DS Subtitle",
+                  "Properties": {},
+                  "Settings": {}
+                }
+                """;
+
+            var dataSourceItems = new RestDataSourceItem("DB Test", new DataSource { Title = "JSON DS", Subtitle = "JSON DS Subtitle" })
+            {
+                Id = "RestItem",
+                Title = "Rest DS Item",
+                Fields = new List<IField>
+                {
+                    new TextField("_id"),
+                    new TextField("name"),
+                }
+            };
+
+            var document = new RdashDocument("My Dashboard");
+            document.Visualizations.Add(new GridVisualization("Test List", dataSourceItems).SetColumns("name"));
+            var expectedJObject = JObject.Parse(expectedJson);
+
+            // Act
+            RdashSerializer.SerializeObject(document);
+            var json = document.ToJsonString();
+            var jObject = JObject.Parse(json);
+            var actualJObject = jObject["DataSources"].LastOrDefault();
```

Comment:
my comment that you resolved was about how two data sources are created when using Rest. This test should check to make sure both are correct.